web/src/components/filings-calendar/tax-access-modal/TaxAccessStepTwo.tsx

```diff
@@ -175,15 +175,25 @@ export const TaxAccessStepTwo = (props: Props): ReactElement => {
           displayName: record?.townDisplayName || profileData.municipality?.displayName || "",
         };
 
-        await updateQueue
-          .queue(userDataToSet)
-          .queueProfileData({
-            taxId: profileData.taxId,
-            encryptedTaxId: encryptedTaxId,
+        await updateQueue.queue(userDataToSet).queueProfileData({
+          taxId: profileData.taxId,
+          encryptedTaxId: encryptedTaxId,
+          municipality: municipalityToSet,
+        });
+
+        if (displayBusinessName() && userDataToSet.taxFilingData.state === "SUCCESS") {
+          await updateQueue.queueProfileData({
+            businessName: profileData.businessName,
+          });
+        }
+
+        if (displayResponsibleOwnerName() && userDataToSet.taxFilingData.state === "SUCCESS") {
+          await updateQueue.queueProfileData({
             responsibleOwnerName: profileData.responsibleOwnerName,
-            municipality: municipalityToSet,
-          })
-          .update();
+          });
+        }
```

Comment:
I agree with pulling out `userDataToSet.taxFilingData.state === "SUCCESS"` a level- debated bringing it up myself but decided against because it was only two conditions but it probably would be cleaner overall. Properly articulates the logical flow of "Did we get a response? If so, do the correct data setting with it"

Reply:
@jphechter I don't think we should be spreading profileData. The point of the updateQueue is that you should only put into it the specific information you want to be updated. I would argue queuing the profileData as a whole makes it less clear what we're actually updating and makes us vulnerable to bugs from inconsistencies between `profileData` and `userData.profileData`

Reply:
@jphechter I agree with @SirSamuelJoseph but additionally there is behavior to not save the businessName if the tax filing state is failed so we need to have that additional if statement and because of that I ended up separating it as I did but pulling the tax filing state to the top level is a good idea, going to make that change.

Reply:
Totally makes sense. Thank you, guys!